plugins/woocommerce-blocks/docs/third-party-developers/extensibility/register-product-collection.md

```diff
@@ -0,0 +1,235 @@
+    },
+    displayLayout: {
+      type: "grid",
+      columns: 3,
+      shrinkColumns: true,
+    },
+  },
+});
+```
+
+Here is how it will look like:
+![image](https://github.com/woocommerce/woocommerce/assets/16707866/5fc1aa20-552a-4e09-b811-08babab46665)
+
+### Example 3: Advanced usage of preview
+
+As you can see below, it's also possible to use `setPreviewState` to set the preview state. In the example below, we are setting `initialPreviewState` and using `setPreviewState` to change the preview state after 5 seconds.
+
+**This example shows:**
+
+- How to access current attributes and location in the preview state
+- How to use async operations
+   	- We are using `setTimeout` to change the preview state after 5 seconds. You can use any async operation, like fetching data from an API, to decide the preview state.
+- How to use the cleanup function as a return value
+   	- We are returning a cleanup function that will clear the timeout after the component is unmounted. You can use this to clean up any resources you have used in `setPreviewState`.
+
+```tsx
+__experimentalRegisterProductCollection({
+  name: "woocommerce/product-collection/my-custom-collection-with-advanced-preview",
+  title: "My Custom Collection with Advanced Preview",
+  icon: "games",
+  description: "This is a custom collection with advanced preview.",
+  keywords: [
+    "My Custom Collection with Advanced Preview",
+    "product collection",
+  ],
+  preview: {
+    setPreviewState: ({
+      setState,
+      attributes: currentAttributes,
+      location,
+    }) => {
+      // setPreviewState has access to the current attributes and location.
+      // console.log( currentAttributes, location );
+
+      const timeoutID = setTimeout(() => {
+        setState({
+          isPreview: false,
+          previewMessage: "",
+        });
+      }, 5000);
+
+      return () => clearTimeout(timeoutID);
+    },
+    initialPreviewState: {
+      isPreview: true,
+      previewMessage:
+        "This is a preview message for my custom collection with advanced preview.",
+    },
+  },
+});
+```
+
+### Example 4: Collection with inner blocks
+
+As you can see below, it's also possible to define inner blocks for the collection. In the example below, we are defining inner blocks for the collection.
+
+```tsx
+__experimentalRegisterProductCollection({
+  name: "woocommerce/product-collection/my-custom-collection-with-inner-blocks",
+  title: "My Custom Collection with Inner Blocks",
+  icon: "games",
+  description: "This is a custom collection with inner blocks.",
+  keywords: ["My Custom Collection with Inner Blocks", "product collection"],
+  innerBlocks: [
+    [
+      "core/heading",
+      {
+        textAlign: "center",
+        level: 2,
+        content: "Title of the collection",
+      },
+    ],
+    [
+      "woocommerce/product-template",
+      {},
+      [
+        ["woocommerce/product-image"],
+        [
+          "woocommerce/product-price",
```

Comment:
What about a product-title? One that has the `isLink` attribute set to true?

Reply:
I am not 100% sure about this. As we are using the `core/post-title` block variation, it requires some opinionated attributes. It's possible that we may replace it with our own block in future. Therefore, I don't want to add it. Also, I wanted to keep the example simple. But I can add it if you think it's needed. Let me know your thoughts. (For more info, you may check this slack discussion - p1719259564963079-slack-C05UMTA7W0P)

Here is how we are currently using it:

```
[
	'core/post-title',
	{
		textAlign: 'center',
		level: 3,
		fontSize: 'medium',
		style: {
			spacing: {
				margin: {
					bottom: '0.75rem',
					top: '0',
				},
			},
		},
		isLink: true,
		__woocommerceNamespace: PRODUCT_TITLE_ID,
	},
],
```

Reply:
Agreed. I see there is more work to be done here and it's probably best not to mix things up. Resolved